src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -0,0 +1,209 @@
+
+    return FORMATTING_FUNCTIONS.containsKey(getName(element)) &&
+           ("fmt".equals(containingFile.getImportPath()) ||
+            "log".equals(containingFile.getImportPath()));
+  }
+
+  @Nullable
+  private static String getName(@NotNull GoFunctionOrMethodDeclaration element) {
+    String name = element.getName();
+    if (name == null) {
+      return null;
+    }
+
+    return name.toLowerCase(Locale.getDefault());
+  }
+
+  private static int getPlaceholders(@Nullable GoExpression argument) {
+    if (argument == null) {
+      return -1;
+    }
+
+    String placeholderText = resolve(argument);
+    if (placeholderText == null) {
+      return -1;
+    }
+
+
+    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(placeholderText);
+    Matcher countPlaceholders;
+
+    int indexCount = 0;
+    int maxIndexCount = 0;
+
+    while (placeholders.find()) {
+      indexCount++;
+
+      countPlaceholders = INDEXED_PLACEHOLDER_PATTERN.matcher(placeholders.group());
+      while (countPlaceholders.find()) {
+        int index = StringUtil.parseInt(countPlaceholders.group().replaceAll("\\[|\\]", ""), -1);
+        if (index > 0 && index != indexCount) {
+          indexCount = index;
+        }
+        if (maxIndexCount < indexCount) {
+          maxIndexCount = indexCount;
+        }
+      }
+      if (maxIndexCount < indexCount) {
+        maxIndexCount = indexCount;
+      }
+    }
+
+    return maxIndexCount;
+  }
+
+  @Nullable
+  private static String resolve(GoExpression argument) {
+    if (argument instanceof GoStringLiteral) {
+      return argument.getText();
+    }
+
+    if (!(argument instanceof GoReferenceExpression)) {
+      return null;
+    }
+
+    PsiReference resolved = argument.getReference();
+    if (resolved == null) {
+      return null;
+    }
+
+    PsiElement element = resolved.resolve();
+    while (element != null &&
+           element.getReference() != null &&
+           element.getReference().resolve() != null) {
+      resolved = element.getReference();
+      element = resolved.resolve();
+    }
+
+    if (element == null) {
+      return null;
+    }
+
+    if (element.getReference() != null) {
+      resolved = element.getReference();
+      if (resolved instanceof GoVarReference) {
+        if (resolved.getElement().getParent() instanceof GoShortVarDeclaration) {
+          element = resolved.getElement();
+          GoShortVarDeclaration shortVarDeclaration = (GoShortVarDeclaration)resolved.getElement().getParent();
+          int varPosition = shortVarDeclaration.getVarDefinitionList().indexOf(element);
+          return resolve(shortVarDeclaration.getExpressionList().get(varPosition));
+        }
+        return "";
+      }
+    }
+    else if (element instanceof GoConstDefinition) {
+      if (element.getParent() instanceof GoConstSpec) {
+        GoConstSpec constSpec = (GoConstSpec)element.getParent();
+        int constPosition = constSpec.getConstDefinitionList().indexOf(element);
```

Comment:
Avoid using indecies without checks. Here is the code, it produces an exception:
```
const s
  	fmt.Printf(s, 1, 2)
```